resources/systemsetup.rb

```diff
@@ -1,26 +1,32 @@
+require 'colorize'
+
 resource_name :systemsetup
-default_action :run
 
-BASE_COMMAND = '/usr/sbin/systemsetup'.freeze
+property :setting, desired_state: false
+property :value, desired_state: true, coerce: proc { |m| m.to_s.capitalize unless m.to_s.include?('/') }
 
-property :get, Array
-property :set, Hash
+default_action :set
 
-action :run do
-  if new_resource.get
-    new_resource.get.each do |flag|
-      execute BASE_COMMAND do
-        command "#{BASE_COMMAND} -get#{flag}"
-        live_stream true
-      end
-    end
+load_current_value do |desired|
+  command_output = shell_out('/usr/sbin/systemsetup', "-get#{desired.setting}").stdout.split(': ').last.strip
```

Comment:
This is the dangerous kind of method chaining that often results in `NoMethodError`s on `nil` objects at runtime. As a general rule, we should try to adhere to the Law of Demeter by only chaining methods on objects of the same type i.e. `some_string.downcase.strip.squeeze.split('//')` is okay because every method in the chain until the last one returns another string. And strictly speaking, this is not the command output; the command output ends at `.stdout`.